### ISSUE TITLE
__filename should be declarable in modules

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#281]
In Node.js scripts, `__filename` is declared by Node.js and cannot be redeclared. In Node.js EJS modules, `__filename` can be declared by the user.

Example:
```js
import url from "url";
let __filename = url.fileURLToPath(import.meta.url); // should be legal, but quick-lint-js complains
let __dirname = path.dirname(__filename); // should be legal, but quick-lint-js complains
```

Make quick-lint-js not complain about declarations of `__filename` or `__dirname` inside of modules.

**Comment by strager** [comment]
`module` has a similar problem. `import module from "module";` is valid, but quick-lint-js complains.

**Comment by strager** [comment]
As a short-term workaround, we should make these variables shadowable.

**Comment by strager** [comment]
Fixed in commit f77cecb150dcc7d3e4d14a37f1f29bbb1e1f358a.
